### DISCUSSION TITLE
[Enhancement suggestion] TanStack (react-query/vue-query) compatibility

### DISCUSSION THREAD

**Discussion by p-mathias** [odata2ts/odata2ts#219] [Ideas]
in react/vue it's common to use react/vue-query (got renamed to TanStack), https://tanstack.com/query/latest/
it would be nice to be able to throw odata2ts fetch/axios into it,
to handle state/mutation.

**Comment by texttechne** [comment]
Hi @p-mathias,

I'm not familiar with this library. A quick glance at it would suggest compatibility with odata2ts in the sense that you would provide the query function using odata2ts there....

Could you expand a bit more how you imagine the integration scenario?

**Comment by texttechne** [comment]
Hi @p-mathias,

converted this to a discussion, feels more approriate than being an issue...

Revisited the library a little bit longer, but actually the integration looks pretty straigth-forward.
Along the lines of the quick start guide:
```ts
const { isLoading, error, data } = useQuery({
    queryKey: ['repoData'],
    queryFn: () => mainODataService.people().query().then(response => response.data.d.results)
  })
```

What do you think? Am I missing something? Are there more intricate problems with that approach?

**Reply by jjspierx** [reply]
I am using odata2ts with tanstack query and it works great.  Tanstack query is designed to work with any data querying library that returns a promise.
